src/deadlineresolver.js

```diff
@@ -80,8 +81,13 @@ class DeadlineResolver extends events.EventEmitter {
       loops.push(this.poll());
     }
     // Create promise that we're done looping
-    this.done = Promise.all(loops).catch((err) => {
-      this.emit('error', err); // This should crash the process
+    this.done = Promise.all(loops).catch(async (err) => {
+      console.log("Crashing the process: %s, as json: %j", err, err);
+      // TODO: use this.monitor.reportError(err); when PR lands:
+      // https://github.com/taskcluster/taskcluster-lib-monitor/pull/27
+      await this.monitor.reportError(err, 'error', {}, true);
+      // Crash the process
+      process.exit(1);
```

Comment:
yeah, that would be nice... It won't be in this PR though...
Ideally, some generic way of constructing a process that polls from azure queue and operates on the messages...